### PR TITLE
Heedls 612a limit admin view in group courses

### PR DESCRIPTION
**Pull request by JBsoftwire**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-612

### Description
The group courses list is now filtered by admin course category ID.

### Screenshots
N/A

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs [hunk 1]
```diff
@@ -176,7 +185,10 @@ public IActionResult GroupCourses(int groupId, int page = 1)
                 return NotFound();
             }
 
-            var groupCourses = groupsDataService.GetGroupCourses(groupId, centreId);
+            var adminId = User.GetAdminId()!.Value;
+            var adminUser = userDataService.GetAdminUserById(adminId)!;
+
+            var groupCourses = groupsService.GetGroupCoursesForCategory(groupId, centreId, adminUser.CategoryIdFilter);
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs:191]: Is there a reason to use this over User.GetAdminCategoryId()?

**Review by AlexJacksonDS — COMMENTED** [review]
Just one question. Not sure if it should be changed or not.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs (hunk 1 above)
Comment (JBsoftwire) [reply DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs:191]: This property handles the conversion of category 0 to null, which is a convention we're using for handling admin category restrictions in SQL. I'll DM you a link to the recent slack conversation.

Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DelegateGroupsController.cs:191]: Dan is making a change to the claim in HEEDLS-564 https://github.com/TechnologyEnhancedLearning/DLSV2/pull/698 (tech lead review). Will move this over to tech lead review for now, but if that does get merged, I think we should update to using the claims version.

**Review by DanBloxham-sw — APPROVED** [review]
I think this good.

I second Alex's comment about the new change to the admin category id claim method, it would be best to use that here. But, at time of writing, it still needs to be merged into master. I think it just requires tech lead re-review on 1 comment, so it might not take long. But if this gets merged before 564, I'll make sure to implement that change here unless you object.

**Pull request merged**